libraries/cmake/source/smartmontools/CMakeLists.txt

```diff
@@ -49,6 +44,10 @@ function(smartmontoolsMain)
 
   if(DEFINED PLATFORM_LINUX)
     target_sources(thirdparty_smartmontools PRIVATE
+      "${library_root}/dev_areca.cpp"
+      "${library_root}/getopt/getopt1.c"
```

Comment:
I would separate C files in their own target, so that we can apply `thirdparty_c_settings` only to them.